src/ring_theory/polynomial/eisenstein.lean

```diff
@@ -185,3 +187,79 @@ end is_domain
 end is_eisenstein_at
 
 end polynomial
+
+
+section is_integral
+
+variables {S : Type w} (K : Type v) (L : Type z) {p : R}
+variables [comm_ring R] [comm_ring S] [algebra R S] [field K] [field L]
+variables [algebra K L] [algebra R L] [algebra R K] [is_scalar_tower R K L] [is_separable K L]
+variables [is_domain R] [normalized_gcd_monoid R] [is_fraction_ring R K] [is_integrally_closed R]
+
+local notation `𝓟` := submodule.span R {p}
+
+open is_integrally_closed power_basis nat polynomial
+
+/-- Let `K` be the field of fraction of an integrally closed domain `R` and let `L` be a separable
+extension of `K`, generated by an integral power basis `B` such that the minimal polynomial of
+`B.gen` is Eisenstein. Given `z : L` integral over `R`, if `Q : polynomial R` is such that
+`(aeval B.gen) Q = p • z`, then `p ∣ Q.coeff 0`. -/
+lemma dvd_coeff_zero_of_aeval_eq_prime_smul_of_minpoly_is_eiseinstein_at {B : power_basis K L}
```

Comment:
Also the other coefficients are divisible by `p`, and in particular `z` is in `adjoin R B.gen`. This is the first step of the induction. It is already quite long, so I didn't PRed the full result yet.